src/PromiseInterface.php

```diff
@@ -44,8 +47,7 @@ public function then(?callable $onFulfilled = null, ?callable $onRejected = null
      * Additionally, you can type hint the `$reason` argument of `$onRejected` to catch
      * only specific errors.
      *
-     * @param callable $onRejected
-     * @return PromiseInterface
+     * @return PromiseInterface<T>
```

Comment:
Shouldn't this be `@return PromiseInterface<TFulfilled>`? (See `then()` above)

Reply:
Will make sure those two align